### PR TITLE
[JBTM-3469] lra-coordinator is now packaged/installed both as WAR and as JAR

### PR DESCRIPTION
**Pull request by jmfinelli**
This PR fixes https://issues.redhat.com/browse/JBTM-3469

Actived test profile:
LRA

Deactivated test profiles:
!MAIN !CORE !TOMCAT !AS_TESTS !RTS !JACOCO !XTS !QA_JTA !QA_JTS_JACORB !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !BLACKTIE !PERF !NO_WIN !DB_TESTS !mysql !db2 !postgres !oracle

### CODE REVIEW

**Review by xstefank — COMMENTED** [review]
Personally, I don't see any issue with approach right now. However, I don't understand how narayana CI passed when this https://github.com/jbosstm/narayana/pull/1822/checks?check_run_id=2401939792 is relevant test issue with this PR.

**Review by ochaloup — APPROVED** [review]
To me the approach is fine. If the war file works as expected then this fix is wanted.

Regarding the testing. The github actions seem to fail again because of the maven dependency download failure.
The `lra-coordinator` was not build and LRA test then failed.

```
...
2021-04-21T15:43:43.2071245Z [ERROR] Failed to execute goal on project arjuna: Could not resolve dependencies for project org.jboss.narayana.arjunacore:arjuna:jar:5.11.2.Final-SNAPSHOT: Could not transfer artifact io.netty:netty-resolver:jar:4.1.48.Final from/to central (https://repo1.maven.org/maven2): Connection reset -> [Help 1]
2021-04-21T15:43:43.2073119Z [ERROR] 
2021-04-21T15:43:43.2073951Z [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2021-04-21T15:43:43.2075000Z [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2021-04-21T15:43:43.2075581Z [ERROR] 
2021-04-21T15:43:43.2076390Z [ERROR] For more information about the errors and possible solutions, please read the following articles:
2021-04-21T15:43:43.2077749Z [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
2021-04-21T15:43:43.2078718Z [ERROR] 
2021-04-21T15:43:43.2079349Z [ERROR] After correcting the problems, you can resume the build with the command
2021-04-21T15:43:43.2080212Z [ERROR]   mvn <goals> -rf :arjuna
....
2021-04-21T15:49:58.3524192Z [INFO] Downloading from redhat: https://maven.repository.redhat.com/ga/org/picketbox/picketbox/5.0.3.Final-redhat-00006/picketbox-5.0.3.Final-redhat-00006.jar
2021-04-21T15:49:58.4377699Z [INFO] Downloaded from redhat: https://maven.repository.redhat.com/ga/org/picketbox/picketbox/5.0.3.Final-redhat-00006/picketbox-5.0.3.Final-redhat-00006.jar (962 kB at 11 MB/s)
2021-04-21T15:49:58.4398431Z [INFO] 
2021-04-21T15:49:58.4406299Z [INFO] ------------------------------------------------------------------------
2021-04-21T15:49:58.4412922Z [INFO] Skipping Thorntail LRA Coordinator
2021-04-21T15:49:58.4419366Z [INFO] This project has been banned from the build due to previous failures.
2021-04-21T15:49:58.4438045Z [INFO] ------------------------------------------------------------------------
2021-04-21T15:49:58.4443114Z [INFO] 
2021-04-21T15:49:58.4455742Z [INFO] ------------------------------------------------------------------------
2021-04-21T15:49:58.4460248Z [INFO] Skipping Quarkus LRA Coordinator
2021-04-21T15:49:58.4467695Z [INFO] This project has been banned from the build due to previous failures.
2021-04-21T15:49:58.4475143Z [INFO] ------------------------------------------------------------------------
```
I'm going to restart the github actions.

**Comment by ochaloup** [comment]
@jmfinelli what about as part of this PR add a change to narayana.sh for maven execution to `--fail-fast`?
* https://github.com/jbosstm/narayana/blob/master/scripts/hudson/narayana.sh#L534
* https://github.com/jbosstm/narayana/blob/master/scripts/hudson/narayana.sh#L537

**Comment by jmfinelli** [comment]
Thanks @ochaloup, I was looking into that failure and arrived at the same conclusion.

**Comment by xstefank** [comment]
if the action pass, then LGTM too.

**Review by mmusgrov — COMMENTED** [review]
Why can't it be fixed by just providing a web.xml file that defines a servlet in the coordinator-war module. That would be a far less invasive change.

I'd have to test it but I believe this may well break the quarkus extension.

**Comment by jmfinelli** [comment]
I tested another way to solve [this](https://issues.redhat.com/browse/JBTM-3469). Restoring the `JaxRsActivator.class` in `lra-coordinator-jar` works too (which means that we might leave the modules `lra-coordinator-jar` and `lra-coordinator-war` as they are). On the other side, I like the idea to have the WAR and the JAR produced by the same module. In fact, `lra-coordinator-war` just re-packs `lra-coordinator-jar` in a WAR.

@mmusgrov I am not sure what you mean when you say that this PR might break the Quarkus extension as the LRA profile uses that extension while testing. Is there something I am missing?

**Comment by mmusgrov** [comment]
Reverting the removal of JaxRsActivator.java could be an option but I would explore adding a web.xml first.

File: rts/lra/coordinator-war/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <!-- One of the way of activating REST Services is adding these lines, the server is responsible for adding the corresponding servlet automatically. If the src folder, org.jboss.as.quickstarts.rshelloworld.HelloWorld class has the Annotations to receive REST invocation-->
+    <display-name>Narayana LRA Coordinator</display-name>
+
+    <listener>
+        <listener-class>io.narayana.lra.coordinator.api.AppContextListener</listener-class>
+    </listener>
+
+    <servlet>
+        <servlet-name>LRACoordinator</servlet-name>
+        <servlet-class>io.narayana.lra.coordinator.api.Coordinator</servlet-class>
```
Comment (ochaloup) [inline rts/lra/coordinator-war/src/main/webapp/WEB-INF/web.xml:13]: it seems you don't consider the recovery coordinator (https://github.com/jbosstm/narayana/blob/master/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/RecoveryCoordinator.java), do you?

**Review by ochaloup — COMMENTED** [review]
Are the WildFly tests passing with the current change or are there something more needed to get it running?

File: rts/lra/coordinator-war/src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (jmfinelli) [reply rts/lra/coordinator-war/src/main/webapp/WEB-INF/web.xml:13]: good catch, thanks :+1: I am gonna add it

**Comment by jmfinelli** [comment]
> Are the WildFly tests passing with the current change or are there something more needed to get it running?

The `war` artifact gets deployed but the `arq.wildfly` profile still does not pass

File: rts/lra/coordinator-war/src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (jmfinelli) [reply rts/lra/coordinator-war/src/main/webapp/WEB-INF/web.xml:13]: After a bit of digging, I realised that the recovery coordinator does not need to be declared as a servlet. `Coordinator` is already doing it for us ([here](https://github.com/jbosstm/narayana/blob/9e4cbbc67025a684230db46fb6b0de21f69773dc/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/Coordinator.java#L149-L152))

**Review by ochaloup — APPROVED** [review]
The new version of the changes seem fine to me. Thanks.

**Review by xstefank — APPROVED** [review]

**Review by mmusgrov — APPROVED** [review]
Thanks Manuel

**Pull request merged**